server/src/main/java/org/elasticsearch/search/MultiValueMode.java

```diff
@@ -389,6 +388,81 @@ protected int pick(SortedDocValues values, DocIdSetIterator docItr, int startDoc
             }
             return ord;
         }
+    },
+
+    /**
+     * Pick the first of all the values
+     */
+    FIRST {
+        @Override
+        protected long pick(SortedNumericDocValues values) throws IOException {
+            return values.nextValue();
```

Comment:
`SortedNumericDocValues` does not preserve the order of the source array, values are sorted during indexing so `first` here would be equivalent to `min`. We need to find a way to allow this `mode` only when it is applied to nested documents and not to select a value on a multi valued field.